### DISCUSSION TITLE
Mat vs Emgu.CV.Image class

### DISCUSSION THREAD

**Discussion by onslauth** [emgucv/emgucv#750] [Q&A]
Hi,

I would like to know if the `Mat` class is managed or not. I tried to read through the available documentation, but all I could really find was the following: [Working with images](https://www.emgu.com/wiki/index.php/Working_with_Images)

It states that `Image` will automatically clean itself up, but does not mention anything regarding the `Mat` class. If the `Mat` class is not managed, what is the correct way to dispose of the object?

Also which would be the most efficient, in terms of processing speed and computation, between `Mat` and `Image`?

**Comment by sn4k3** [comment]
`Mat` is a wrapper around openCV `Mat`.
You should always dispose `Mat`, it's not that hard using modern C#: `using vat mat = new Mat(); // == easy!`. But if you don't it will clean itself when instance is lost and GC does it pass. All C# objects that implements `IDisposable` and properly release unmanaged resources there, will get release once instance is lost, GC will call Dispose on those lost objects.

You should use `Mat` instead `Image`, since Mat is the new way to handle data and also more efficient.
Also `Image` will be removed in future
